1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,208 @@ Bitcoin blockchain, such as exchanges or custodians:
+understanding of how other nodes and miners on the network will behave. There
+are two major difficulties to consider:
+
+1. **DoS** - how do we prevent malicious actors from consuming node resources
+   by repeatedly broadcasting replacement transactions for zero cost?
+2. **Incentive alignment** - how do we make sure that replacement transactions
+   are incentive-compatible and miners want to mine them?
+
+Satoshi's first version of transaction replacement failed on both of these
+counts. The `nSequence` number could be incremented millions of times without
+changing the fee, forcing each node in the network to re-validate all of the
+transactions many times at zero cost to the attacker. The replacement logic also
+didn't consider the transaction fee at all, and would replace an old
+transaction with a new transaction even if the replacement had a lower fee. If
+a miner was running this software, the replacement behaviour would be costing
+him money since he'd be mining transactions with lower fees.
+
+For those reasons, Satoshi [disabled transaction replacement][disable replacement]
+in 2010. When asked later why he'd disabled that code, his response was:
+
+> Just to reduce surface area.  It wouldn't help with increasing tx fee. A tx
+> starts being valid at nLockTime.  It wouldn't work to have a tx that stops
+> being valid at a certain time; once a tx ever becomes valid, it must stay
+> valid permanently.
+
+Replacing a transaction does not invalidate the old transaction, so unless
+there is an incentive for the miner to use the new transaction, then just
+replacing an old transaction with one that has higher `nSequence` numbers
+is not incentive-compatible.
+
+## Replace-by-Fee
+
+Replace-by-fee is an effort to make transactions replaceable whilst avoiding
+the problems of exposing nodes to DoS vulnerabilities and misaligned
+incentives.  This is achieved by only allowing replacement if the replacement
+transaction has a higher fee than the replaced transaction. This prevents DoS
+attacks since every replacement transaction costs the attacker additional
+money, and ensures incentive alignment since the replacement transaction is
+financially advantageous to the miner.
+
+There are several [variants of replace-by-fee][rbf variants]. Almost all nodes
+on the Bitcoin are thought to implement 'opt-in RBF' as defined in [BIP 125][].
+
+Bitcoin Core may implement 'delayed RBF' at some point in the future. There
+is currently an [open PR][delayed rbf pr], which has support from several
+developers.
 
 ## BIP 125
 
-TODO
+BIP 125 is a specification for opt-in RBF written by David Harding and Peter
+Todd in December 2015. It describes the conditions under which an unconfirmed
+transaction in a node's mempool will be replaced. BIP 125 is referred to as
+*opt-in* RBF since the node will only allow a transaction to be replaced if
+the sender has explicitly signalled that he wants the transaction to be
+replacable.
```

Comment:
Please feel free to keep it as-is.  First because, IMO, the primary author of a document should be the one to define its style, and second because it's always seemed to me that one L is wrong---I only do it that way because my spell checker says so.

Reply:
> it might be worth rephrasing it here

done

> Please feel free to keep it as-is

No, I'm going to try to converge on American spelling. Most of my writing is some mid-atlantic mess. It's better to be consistent.